### PR TITLE
Expand path manager coverage.

### PR DESCRIPTION
**Pull request by ossama-othman**

### CODE REVIEW

File: src/path_manager.c [hunk 1]
```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
...
+        mptcpd_token_t const *token;
+
+        /// Local MPTCP address ID.
+        mptcpd_aid_t const *laddr_id;
+
+        /// Remote MPTCP address ID.
+        mptcpd_aid_t const *raddr_id;
+
+        /// Local IPv4 address.
+        in_addr_t laddr4;
```
Comment (mjmartineau) [inline src/path_manager.c:126]: Using the value here (and with `raddr4`) rather than a pointer makes it incompatible with MPTCP_GET_NL_ATTR (leading to the CI build break and the same on my machine).

Converting to a value rather than pointer also means 0.0.0.0 is treated as "no address" rather than "all local addresses", not sure if 0.0.0.0 is used in any events? Could be useful to allow it for future flexibility.

File: src/path_manager.c [hunk 2]
```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
...
+
+        /**
+         * @brief MPTCP subflow related event error.
+         *
+         * @note This is value is equal to the @c sk_err member of
+         *       @c struct @c sock in the Linux kernel.
+         */
+        uint8_t *error;
+
+#ifdef MPTCPD_ENABLE_PM_NAME
```
Comment (mjmartineau) [inline src/path_manager.c:157]: Is this a good time to purge this PM_NAME code?

**Review by mjmartineau — COMMENTED** [review]
I do think it cleans up the code nicely to consolidate the netlink parsing like this. Sparse usage of the attrs struct not a problem as far as I'm concerned.

File: src/path_manager.c (hunk 1 above)
Comment (ossama-othman) [reply src/path_manager.c:126]: > Using the value here (and with `raddr4`) rather than a pointer makes it incompatible with MPTCP_GET_NL_ATTR (leading to the CI build break and the same on my machine).

Right.  I was going to modify `MPTCP_GET_NL_ADDR` to return a value to address this build error but I'll hold off on that given your comment below.

> Converting to a value rather than pointer also means 0.0.0.0 is treated as "no address" rather than "all local addresses", not sure if 0.0.0.0 is used in any events? Could be useful to allow it for future flexibility.

That's fine.  I'll stick with a pointer to `in_addr_t` instead.

File: src/path_manager.c (hunk 2 above)
Comment (ossama-othman) [reply src/path_manager.c:157]: Sure.

File: src/path_manager.c [hunk 3]
```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
...
                 case MPTCP_ATTR_LOC_ID:
                 case MPTCP_ATTR_REM_ID:
                 case MPTCP_ATTR_BACKUP:
+                case MPTCP_ATTR_ERROR:
+                case MPTCP_ATTR_FLAGS:
+                case MPTCP_ATTR_TIMEOUT:
                 case MPTCP_ATTR_IF_IDX:
+                case MPTCP_ATTR_RESET_REASON:
+                case MPTCP_ATTR_RESET_FLAGS:
                         // Unused and ignored, at least for now.
```
Comment (ossama-othman) [inline src/path_manager.c:229]: We should probably retrieve at least some of these attributes, such as `MPTCP_ATTR_IF_IDX`, `MPTCP_ATTR_{LOC,REM}_ID` and `MPTCP_ATTR_BACKUP`.

File: tests/test-plugin.c [hunk 4]
```diff
@@ -173,8 +173,8 @@ static void test_nonexistent_plugins(void const *test_data)
         l_queue_destroy(plugins_list, NULL);
 
         /*
-          Force plugin name-to-ops lookup to fail excercise additional
-          error paths.
+          Exercise additional error paths by forcing plugin
+          name-to-ops lookup to fail.
```
Comment (ossama-othman) [inline tests/test-plugin.c:177]: This rewording change inadvertently made it into this pull request.  Should we leave it in place or move it to a different pull request?

File: src/path_manager.c (hunk 2 above)
Comment (mjmartineau) [reply src/path_manager.c:157]: No strong preference. I suppose it's a separate topic which would make me lean toward a separate PR, but it's not worth lots of extra effort to split it out.

File: src/path_manager.c [hunk 5]
```diff
@@ -85,106 +85,145 @@ static bool validate_attr_len(size_t actual, size_t expected)
...
                         break;
-#ifdef MPTCPD_ENABLE_PM_NAME
-                case MPTCP_ATTR_PATH_MANAGER:
-                        pm_name = get_pm_name(data, len);
+                case MPTCP_ATTR_FAMILY:
+                case MPTCP_ATTR_FLAGS:
+                case MPTCP_ATTR_TIMEOUT:
+                case MPTCP_ATTR_RESET_REASON:
+                case MPTCP_ATTR_RESET_FLAGS:
+                        // Unused and ignored, at least for now.
```
Comment (mjmartineau) [inline src/path_manager.c:201]: Are the remaining unused/ignored attrs populated by the upstream kernel at all?

File: src/path_manager.c [hunk 6]
```diff
@@ -85,106 +85,145 @@ static bool validate_attr_len(size_t actual, size_t expected)
...
                         break;
-#endif  // MPTCPD_ENABLE_PM_NAME
                 default:
-                        l_warn("Unknown MPTCP_EVENT_CREATED "
-                               "attribute: %d",
-                               type);
+                        l_warn("Unknown MPTCP genl attribute: %d", type);
```
Comment (mjmartineau) [inline src/path_manager.c:204]: Should this be a warning or just informational? No harm in running on a newer kernel that might populate additional attributes.

**Review by mjmartineau — COMMENTED** [review]
A couple of questions (see inline comments)

File: src/path_manager.c (hunk 6 above)
Comment (ossama-othman) [reply src/path_manager.c]: Informational is better.  I changed it accordingly.

File: src/path_manager.c (hunk 5 above)
Comment (ossama-othman) [reply src/path_manager.c:201]: As far as I can tell, only `MPTCP_ATTR_RESET_REASON` and  `MPTCP_ATTR_RESET_FLAGS` are populated, both when a subflow closed event is sent.

**Review by mjmartineau — APPROVED** [review]
